File: modules/spring-data/testdata/java/reference/queryInputParameter/UserRepository_resolvable.java
package com.example;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    @Query("SELECT u FROM User u WHERE u.firstName = :name")
    List<User> findByFirstName(@Param("name") String firstName);

    @Query("SELECT u FROM User u WHERE u.lastName = :lastName")
    List<User> findByLastName(String lastName);

    @Query("SELECT u FROM User u WHERE u.age >= ?1")
    List<User> findByAgeGreaterThan(int age);
}

File: modules/spring-data/testdata/kotlin/reference/queryInputParameter/UserRepository_resolvable.kt
package com.example;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
interface UserRepository : JpaRepository<User, Long> {
    @Query("SELECT u FROM User u WHERE u.firstName = :name")
    fun findByFirstName(@Param("name") firstName: String): List<User>

    @Query("SELECT u FROM User u WHERE u.lastName = :lastName")
    fun findByLastName(lastName: String): List<User>

    @Query("SELECT u FROM User u WHERE u.age >= ?1")
    fun findByAgeGreaterThan(age: Integer): List<User>
}
